GoogleChromeLabs/ps-analysis-tool
# Feature: Add settings page and control number of tabs being processed.

## Description
In the base branch `fix/cookie-processing-message` the processing was primarily for a single tab by default. This PR aims to give users an option to switch between the processing of a single tab and all the tabs.

## Relevant Technical Choices
- By default the setting no-restriction is checked and these settings are stored in sync storage as specified in developer documentation.
- While processing a request we check the option and then process a request.

## Testing Instructions

- Pull `feat/settings-page` and then run `npm i && npm run dev`.
- Now open multiple tabs with URLs.
- Now in the popup, you should be able to see all tabs being processed.
- Right-click on the popup icon and select `options` there check the radio button labelled `single tab processing`.
- Now go back to the tabs with the URLs You should see only the first tab being processed, in the other tabs popup icon will show the message `This tool works best with a single tab` and with a button `Analyze this tab`.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/e83a61c2-7385-47c5-9e7d-573726635a34

Fixes #133
